lib/spack/spack/fetch_strategy.py

```diff
@@ -1129,15 +1130,18 @@ def fetch(self):
 
         parsed_url = url_util.parse(self.url)
         if parsed_url.scheme != 's3':
-            raise ValueError(
+            raise FetchError(
                 'S3FetchStrategy can only fetch from s3:// urls.')
 
         tty.msg("Fetching %s" % self.url)
 
         basename = os.path.basename(parsed_url.path)
 
         with working_dir(self.stage.path):
-            _, headers, stream = web_util.read_from_url(self.url)
+            try:
+                _, headers, stream = web_util.read_from_url(self.url)
```

Comment:
(request) I think it might make sense to catch `urllib_error.URLError` inside the `web_util.read_from_url` function (and it could throw a `SpackWebError` which I think would avoid sidestepping the `try/catch` logic in the `Stage.fetch` loop.

Reply:
Good idea! Update incoming.